### ISSUE TITLE
Stash is using unsupported Go version 1.13.x

### ISSUE THREAD

**Issue by stashist** [stashapp/stash#1272]
Stash releases are compiled using version 1.13.15 (see Dockerfile for the stashapp/compiler image).

Support for the 1.13 branch ended on 2020-08-11 with the release of version 1.15.0.

Based on golang-announce there appears to have been at least 4 or more security releases since EoL that we're missing.

Unfortunately simply rebuilding the compiler image from 1.16.3 breaks darwin during the cross compile. (Using 1.15.11 might work; I haven't tried.)

**Comment by bnkai** [comment]
For cross compiling (macos) we need a packaged version of the osx sdk and the relevant clang version.
Unless you replace the 10.11 image in the compiler docker with a newer one and edit the checkout for osxcross the compiler image wont work properly. 
I assume the above from this part of go 1.15 release notes `As announced in the Go 1.14 release notes, Go 1.15 requires macOS 10.12 Sierra or later; support for previous versions has been discontinued.`

**Comment by stashist** [comment]
I took a look at https://firefox-source-docs.mozilla.org/widget/cocoa/sdks.html and ... wow.

My reading of the situation is that the only way to provide macos builds in a sdk license compliant way is to use github's macos runners ([e.g.](https://github.blog/changelog/2020-01-09-github-actions-updates-to-the-macos-10-15-virtual-environment/)). Unless someone volunteers a spare mac as a self-hosted runner ...

I took a look at how we build the current compiler image. Looks like upstream (and their upstream as well) don't support 10.12+ yet (ever?). So there's no great stopgap measure either.
